### PR TITLE
added versioning for migration

### PR DESCRIPTION
**Pull request by YuJuncen**
This PR added the vesrion and creator field to the `Migration` type for troubleshoting.

### CODE REVIEW

**Review by BornChanger — APPROVED** [review]

File: proto/brpb.proto [hunk 1]
```diff
@@ -876,7 +876,17 @@ message DeleteSpansOfFile {
     uint64 whole_file_length = 3;
 }
 
+enum MigrationVersion {
+    // The initial version. Somehow "UNKNOWN".
+    M0 = 0;
```
Comment (Tristan1900) [inline proto/brpb.proto:881]: nit, maybe V0 and V1? usually use V as version,

File: proto/brpb.proto [hunk 2]
```diff
@@ -876,7 +876,17 @@ message DeleteSpansOfFile {
...
+    // Added `creator` and `version`.
+    M1 = 1;
+}
+
 message Migration {
+    string creator = 5;
```
Comment (Tristan1900) [inline proto/brpb.proto:887]: nit, maybe sort in the numeric order, put them to the end of the message.

**Review by Tristan1900 — APPROVED** [review]
LGTM!

File: proto/brpb.proto (hunk 1 above)
Comment (YuJuncen) [reply proto/brpb.proto:881]: protocol buffer requires the enum name is unique among *all* enums in the same package. The variant `V1` was used by `MetaFileVersion`...

File: proto/brpb.proto (hunk 2 above)
Comment (YuJuncen) [reply proto/brpb.proto:887]: I think the `version` and `creator` field should be more primitive and essential than other fields. So I put them in the first line and people may notice them easier.

**Review by 3pointer — APPROVED** [review]

**Pull request merged**